AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -122,7 +122,10 @@ extension Statement: StatementModelConvertible {
             // For example, when the value is the `id` of the Blog, then the field.isPrimaryKey is satisfied.
             // Every association of the Blog, such as the has-many Post is populated with the List with
             // associatedId == blog's id. This way, the list of post can be lazily loaded later using the associated id.
-            if let id = modelValue as? String, field.isPrimaryKey {
+            if let id = modelValue as? String,
+               (field.name == ModelIdentifierFormat.Custom.sqlColumnName || // this is the `@@primaryKey` (CPK)
```

Comment:
let me know if i am understanding correctly - 
(a || b) - 
a is v2 way of checking for PK
b is v1 way of checking for PK
and b is added for backward compatibility to models generated with old CLI version